src/internal/EncoderBase.ts

```diff
@@ -0,0 +1,25 @@
+import * as Enc from 'io-ts/Encoder'
+import { Schemable2 } from './Schemable2'
+
+export { URI } from 'io-ts/Encoder'
+
+export const Schemable: Schemable2<Enc.URI> = {
+  URI: Enc.URI,
+  literal: () => Enc.id(),
+  string: Enc.id(),
+  number: Enc.id(),
+  nullable: Enc.nullable,
+  // @ts-expect-error -- typelevel difference
+  struct: Enc.struct,
```

Comment:
The definition of `Enc.struct` looks like

```ts
export function struct<P extends Record<string, Encoder<any, any>>>(
  properties: P
): Encoder<{ [K in keyof P]: OutputOf<P[K]> }, { [K in keyof P]: TypeOf<P[K]> }>
```

Since `P` is an uninstantiated generic in this case, TypeScript doesn't resolve it yet, and `OutputOf<P[K]>` and `TypeOf<P[K]>` end up not being assignable to `EofKind2<'io-ts/Encoder', Properties[K]>` and `AofKind2<'io-ts/Encoder', Properties[K]>`